compiler/rustc_next_trait_solver/src/solve/effect_goals.rs

```diff
@@ -54,21 +57,19 @@ where
                     return Err(NoSolution);
                 }
 
-                ecx.probe_trait_candidate(source).enter(|ecx| {
+                return ecx.probe_trait_candidate(source).enter(|ecx| {
                     let assumption_trait_pred = ecx.instantiate_binder_with_infer(host_clause);
                     ecx.eq(
                         goal.param_env,
                         goal.predicate.trait_ref,
                         assumption_trait_pred.trait_ref,
                     )?;
                     then(ecx)
-                })
-            } else {
-                Err(NoSolution)
+                });
             }
-        } else {
-            Err(NoSolution)
         }
+
+        Err(NoSolution)
```

Comment:
please move these spurious cleanups/changes into a separate PR

Reply:
Removed these changes, they were fixup'ed into the wrong commit from later changes in the Part II with constness.